Modules/_zoneinfo.c

```diff
@@ -1457,7 +1457,8 @@ parse_tz_str(PyObject *tz_str_obj, _tzrule *out)
     PyObject *dst_abbr = NULL;
     TransitionRuleType *start = NULL;
     TransitionRuleType *end = NULL;
-    long std_offset, dst_offset;
+    long std_offset = 1 << 20;
+    long dst_offset = 1 << 20;
```

Comment:
Probably want to add a comment about the magic numbers here, like:

```suggestion
    // Initialize offsets to invalid value (> 24 hours)
    long std_offset = 1 << 20;
    long dst_offset = 1 << 20;
```